src/kit/CodeEditor.tsx

```diff
@@ -1,9 +1,5 @@
-import { markdown, markdownLanguage } from '@codemirror/lang-markdown';
-import { python } from '@codemirror/lang-python';
-import { StreamLanguage } from '@codemirror/language';
-import { json } from '@codemirror/legacy-modes/mode/javascript';
-import { yaml } from '@codemirror/legacy-modes/mode/yaml';
-import ReactCodeMirror from '@uiw/react-codemirror';
+import { langs } from '@uiw/codemirror-extensions-langs';
```

Comment:
q: what languages are we importing here?

Reply:
Here's the list.
<details>
<summary>Lang List</summary>

```ts
export declare const langs: {
    apl: () => StreamLanguage<unknown>;
    asciiArmor: () => StreamLanguage<unknown>;
    asterisk: () => StreamLanguage<unknown>;
    c: () => StreamLanguage<unknown>;
    csharp: () => import("@codemirror/language").LanguageSupport;
    scala: () => StreamLanguage<unknown>;
    solidity: () => import("@codemirror/language").LanguageSupport;
    kotlin: () => StreamLanguage<unknown>;
    shader: () => StreamLanguage<unknown>;
    nesC: () => StreamLanguage<unknown>;
    objectiveC: () => StreamLanguage<unknown>;
    objectiveCpp: () => StreamLanguage<unknown>;
    squirrel: () => StreamLanguage<unknown>;
    ceylon: () => StreamLanguage<unknown>;
    dart: () => StreamLanguage<unknown>;
    cmake: () => StreamLanguage<unknown>;
    cobol: () => StreamLanguage<unknown>;
    commonLisp: () => StreamLanguage<unknown>;
    crystal: () => StreamLanguage<unknown>;
    cypher: () => StreamLanguage<unknown>;
    d: () => StreamLanguage<unknown>;
    diff: () => StreamLanguage<unknown>;
    dtd: () => StreamLanguage<unknown>;
    dylan: () => StreamLanguage<unknown>;
    ebnf: () => StreamLanguage<unknown>;
    ecl: () => StreamLanguage<unknown>;
    eiffel: () => StreamLanguage<unknown>;
    elm: () => StreamLanguage<unknown>;
    factor: () => StreamLanguage<unknown>;
    fcl: () => StreamLanguage<unknown>;
    forth: () => StreamLanguage<unknown>;
    fortran: () => StreamLanguage<unknown>;
    gas: () => StreamLanguage<unknown>;
    gherkin: () => StreamLanguage<unknown>;
    groovy: () => StreamLanguage<unknown>;
    haskell: () => StreamLanguage<unknown>;
    haxe: () => StreamLanguage<unknown>;
    http: () => StreamLanguage<unknown>;
    idl: () => StreamLanguage<unknown>;
    jinja2: () => StreamLanguage<unknown>;
    mathematica: () => StreamLanguage<unknown>;
    mbox: () => StreamLanguage<unknown>;
    mirc: () => StreamLanguage<unknown>;
    modelica: () => StreamLanguage<unknown>;
    mscgen: () => StreamLanguage<unknown>;
    mumps: () => StreamLanguage<unknown>;
    nsis: () => StreamLanguage<unknown>;
    ntriples: () => StreamLanguage<unknown>;
    octave: () => StreamLanguage<unknown>;
    oz: () => StreamLanguage<unknown>;
    pig: () => StreamLanguage<unknown>;
    properties: () => StreamLanguage<unknown>;
    protobuf: () => StreamLanguage<unknown>;
    puppet: () => StreamLanguage<unknown>;
    q: () => StreamLanguage<unknown>;
    sas: () => StreamLanguage<unknown>;
    sass: () => import("@codemirror/language").LanguageSupport;
    liquid: () => import("@codemirror/language").LanguageSupport;
    mermaid: () => import("@codemirror/language").LanguageSupport;
    nix: () => import("@codemirror/language").LanguageSupport;
    svelte: () => import("@codemirror/language").LanguageSupport;
    sieve: () => StreamLanguage<unknown>;
    smalltalk: () => StreamLanguage<unknown>;
    solr: () => StreamLanguage<unknown>;
    sparql: () => StreamLanguage<unknown>;
    spreadsheet: () => StreamLanguage<unknown>;
    stex: () => StreamLanguage<unknown>;
    textile: () => StreamLanguage<unknown>;
    tiddlyWiki: () => StreamLanguage<unknown>;
    tiki: () => StreamLanguage<unknown>;
    troff: () => StreamLanguage<unknown>;
    ttcn: () => StreamLanguage<unknown>;
    turtle: () => StreamLanguage<unknown>;
    velocity: () => StreamLanguage<unknown>;
    verilog: () => StreamLanguage<unknown>;
    vhdl: () => StreamLanguage<unknown>;
    webIDL: () => StreamLanguage<unknown>;
    xQuery: () => StreamLanguage<unknown>;
    yacas: () => StreamLanguage<unknown>;
    z80: () => StreamLanguage<unknown>;
    wast: typeof wast;
    javascript: typeof javascript;
    jsx: () => import("@codemirror/language").LanguageSupport;
    typescript: () => import("@codemirror/language").LanguageSupport;
    tsx: () => import("@codemirror/language").LanguageSupport;
    vue: () => import("@codemirror/language").LanguageSupport;
    angular: () => import("@codemirror/language").LanguageSupport;
    json: typeof json;
    html: typeof html;
    css: typeof css;
    python: typeof python;
    markdown: () => import("@codemirror/language").LanguageSupport;
    xml: typeof xml;
    sql: typeof sql;
    mysql: () => import("@codemirror/language").LanguageSupport;
    pgsql: () => import("@codemirror/language").LanguageSupport;
    java: typeof java;
    rust: typeof rust;
    cpp: typeof cpp;
    lezer: typeof lezer;
    php: typeof php;
    go: () => StreamLanguage<unknown>;
    shell: () => StreamLanguage<unknown>;
    lua: () => StreamLanguage<unknown>;
    swift: () => StreamLanguage<unknown>;
    tcl: () => StreamLanguage<unknown>;
    yaml: () => StreamLanguage<unknown>;
    vb: () => StreamLanguage<unknown>;
    powershell: () => StreamLanguage<unknown>;
    brainfuck: () => StreamLanguage<unknown>;
    stylus: () => StreamLanguage<unknown>;
    erlang: () => StreamLanguage<unknown>;
    nginx: () => StreamLanguage<unknown>;
    perl: () => StreamLanguage<unknown>;
    ruby: () => StreamLanguage<unknown>;
    pascal: () => StreamLanguage<unknown>;
    livescript: () => StreamLanguage<unknown>;
    less: () => import("@codemirror/language").LanguageSupport;
    scheme: () => StreamLanguage<unknown>;
    toml: () => StreamLanguage<unknown>;
    vbscript: () => StreamLanguage<unknown>;
    clojure: () => import("@codemirror/language").LanguageSupport;
    coffeescript: () => StreamLanguage<unknown>;
    julia: () => StreamLanguage<unknown>;
    dockerfile: () => StreamLanguage<unknown>;
    r: () => StreamLanguage<unknown>;
};
```
</details> 

> Seems to work, my only concern is around whether we're over-importing language definitions. Looking at the react-codemirror extensions, I don't think we need to support PHP or Clojure, for example.

I dont think these are harmful because both build artifact size are almost the same (16.2MB). and i think unused code should be tree shook.

Reply:
Post-note: hew library build size is unaffected because the library excludes dependencies from the bundle. This is necessary to prevent the individual modules from including duplicate instances of the dependencies. As a result, the build size impact isn't seen until the consumer build.

Adding all codemirror languages adds 2 megabytes to the end bundle size -- you can see this comparing the results of the build-react jobs in determined-ai/determined#9853 and determined-ai/deterimined#9850. Note: **this code can't be tree shaken** -- the package is set up in a way that it imports all languages and there's no way to tell the compiler which languages the consumer is using.  Also, because the code editor in the app maps file to mode based on file extension, adding these does not add the language support to the frontend.